src/main/java/org/cyclops/integrateddynamics/core/network/diagnostics/GuiNetworkDiagnostics.java

```diff
@@ -263,67 +285,83 @@ public void mouseClicked(MouseEvent e) {
+                                    ObservableObserverData observerData = getObserverDataFromRow(row);
+                                    if (observerData != null && NetworkDiagnosticsPartOverlayRenderer.getInstance().hasPartPos(observerData.toPartPos())) {
                                         c.setBackground(Color.CYAN);
                                     } else {
-                                        c.setBackground(isSelected ? Color.BLUE : Color.WHITE);
+                                        c.setBackground(isCellSelected(row, column) ? getSelectionBackground() : null);
                                     }
                                     return c;
                                 }
-                            });
-                            tableObservers.getColumnModel().removeColumn(tableObservers.getColumn("_id"));
-                            tableObservers.setAutoCreateRowSorter(true);
+                            };
+
                             tableObservers.addMouseListener(new MouseAdapter() {
                                 @Override
                                 public void mouseClicked(MouseEvent e) {
                                     super.mouseClicked(e);
                                     if (e.getClickCount() == 1) {
                                         JTable target = (JTable)e.getSource();
                                         int row = target.rowAtPoint(e.getPoint());
-                                        ObservableObserverData partData = getObserverDataFromRow(row);
-                                        if (partData != null) {
-                                            teleportPlayer(e, partData.toPartPos());
+                                        ObservableObserverData observerData = getObserverDataFromRow(row);
+                                        if (observerData != null) {
+                                            teleportPlayer(e, observerData.toPartPos());
+                                            ((DefaultTableModel) target.getModel()).fireTableRowsUpdated(row, row);
                                         }
                                     }
                                 }
                             });
 
+                            for (JTable table : new JTable[] { tableParts, tableObservers }) {
+                                table.getColumnModel().removeColumn(table.getColumn("_id"));
+                                table.setAutoCreateRowSorter(true);
+
+                                table.setDefaultRenderer(Double.class, new DefaultTableCellRenderer() {
+                                    @Override
+                                    public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
+                                        //noinspection MalformedFormatString
```

Comment:
Just casting to `Double` makes this unnecessary.

Reply:
But then IntelliJ will complain about the superfluous cast :)

So, since it complains in both cases, I picked the alternative with less casts.